### ISSUE TITLE
Add capability for returning list of security groups matching filter

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#1252]
_This issue was originally opened by @beakerman29 as hashicorp/terraform#15640. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Terraform Version 0.8.8

Content's of TF file:

```hcl
provider "aws" {
    shared_credentials_file = "/terraform/credentials"
      profile = "${var.profile}"
      region = "${var.regions_east}"
}
```

# query current security ids for elb

```hcl
data "aws_security_group" "security_group" {
    filter {
        name = "group-name"
        values = ["stack-elb-*"] 
    }
}
```

### Expected Behavior
What should have happened?

the data should return a list of security groups matching the filter value

### Actual Behavior
What actually happened?

errors out and complains that the security group has to match specifically one group

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. terraform plan

### Important Factoids
Nothing unusual

**Comment by Anonymouslemming** [comment]
I found this while trying to do something similar today. It would be very useful to be able to get a list of security group IDs instead of a single security group.

**Comment by kmcquade** [comment]
Yep. I could really use this

**Comment by iamhsa** [comment]
I think that would be very welcome.

**Comment by thakkerdhawal** [comment]
+1

**Comment by bflad** [comment]
Turns out, this was actually released today! A new `aws_security_groups` data source (note the plural `s`) is now available in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by thakkerdhawal** [comment]
thanks
